scheduler/test/cook/test/util.clj

```diff
@@ -56,4 +56,15 @@
     (is (= @state {}))
     (is (= (set-atom! state "a") {}))
     (is (= (set-atom! state {:a :b}) "a"))
-    (is (= @state {:a :b}))))
\ No newline at end of file
+    (is (= @state {:a :b}))))
+
+(deftest test-format-map-for-structured-logging
+  "Tests that the format-map-for-structured logging preserves nested maps."
+  (let [map {:integer 2 :float 1.2 :string "foo" :nested-map {:nested-string "bar" :nested-int 3}}
+        formatted-map (format-map-for-structured-logging map)]
+    (is (number? (:integer formatted-map)))
```

Comment:
Why not also do a straight:

`(is (eq {.....}  formattted-map)`

Also, include a UUID?